tabbycat/notifications/views.py

```diff
@@ -125,14 +125,16 @@ def post(self, request, *args, **kwargs):
 
         data = json.loads(request.body)
 
-        records = SentMessage.objects.filter(hook_id__in=[obj['hook-id'] for obj in data if 'hook-id' in obj])
+        # Ignore all objects without a Tabbycat-specified hook ID
+        data = [obj for obj in data if 'hook-id' in obj and obj['hook-id'] is not None]
+        records = SentMessage.objects.filter(hook_id__in=[obj['hook-id'] for obj in data])
         record_lookup = {smr.hook_id: smr.id for smr in records}
         statuses = []
 
         for obj in data:
             dt = datetime.fromtimestamp(obj['timestamp'])
             timestamp = timezone.make_aware(dt, timezone.utc)
-            email_id = record_lookup.get(obj.get('hook-id'))
+            email_id = record_lookup.get(obj['hook-id'], None)
             if email_id is None:
```

Comment:
Wouldn't need this check then, right?

Reply:
Not the inner one! But still need the outer one: it could be a hook ID that wasn't in the database (so it'll still be in `data`, but won't be in `record_lookup`).